lib/rack/ratelimit.rb

```diff
@@ -163,5 +175,23 @@ def increment(classification, timestamp)
         end
       end
     end
+
+    class RedisCounter
+      def initialize(redis, name, period)
+        @redis, @name, @period = redis, name, period
+      end
+
+      # Increment the request counter and return the current count.
+      def increment(classification, timestamp)
+        key = 'rack-ratelimit/%s/%s/%i' % [@name, classification, timestamp]
+
+        # Returns [count, expire_ok] response for each multi command.
+        # Return the first, the count.
+        @redis.multi do |redis|
+          redis.incr key
+          redis.expire key, @period
```

Comment:
Correct. By setting the expiry unconditionally, we don't have to care about the current key state. Once we move to the next timeslice, the key will change and the old key will naturally expire.